compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -1361,6 +1431,9 @@ static LogicalResult setElementwiseGenericOpRootConfig(
       getDefaultDistributedLevelTileSizes(genericOp, minTileSizes, maxTileSizes,
                                           /*allowIncompleteTile=*/true);
 
+  // TODO(dcaballe): The logic below is disconnected from the main tile size
+  // selection logic in getMaxTileSize. We should either port it there or remove
+  // it.
```

Comment:
This is hard to port it to getMaxTileSize because it needs more context. The intention is avoiding tiny dispatches. I.e., we want the execution time of each dispatch launch is big enough (like 100us-500us). This is only applied to elementwise operations because it is usually cheap. 

IMO, the better plan is having a cost model about execution time and propagate it to default flow tiling sizes selection. It is a bottom-up approach while today we're using top-down approach.

(It's not related the PR. I point it out because I see the comment. :p)

Reply:
It should be easy to look at the ops within a generic op and get a sense of the computation density vs memory. I think if some specialization like this is necessary, getMin/getMax tile size functions should be extended accordingly, even if that means ignoring those extensions for ops that don't need them. I personally found really confusing that we bypass the current "tile selection infra" (i.e., the aforementioned functions) for this specific case.

Reply:
I dont think we really need to look much at the region of the `linalg.generic`. I assume most generic ops are bandwidth bound, and indexing maps gives us most of the information we need here.

Reply:
Well... we have some monster linalg.generic ops with tosa.resize and friends. I don't think we can always assume that they are bandwidth bound. This information should be trivial to get.